cpu.go

```diff
@@ -254,14 +258,25 @@ func (ca *Cagent) CPUWatcher() *CPUWatcher {
 
 	stat.UtilAvg.SetDurationsMinutes(durations...)
 	stat.UtilAvg.mu.Unlock()
+	ca.cpuWatcher = &stat
+
+	if len(ca.CPUUtilTypes) > 0 && len(ca.CPUUtilDataGather) > 0 || len(ca.CPULoadDataGather) > 0 {
+		// optimization to prevent CPU watcher to run in case CPU util metrics not are not needed
+		err := stat.Once()
+		if err != nil {
+			log.Error("[CPU] Failed to read utilisation metrics: " + err.Error())
+		} else {
+			go stat.Run()
+		}
+	}
 
-	return &stat
+	return ca.cpuWatcher
```

Comment:
Since line 264 does set the value of ca.cpuWatcher we don't necessarily have to return the value again here I think. Except this makes using it easier in other places.
There are a couple of possible error cases, shouldn't we return the error in these cases?

Reply:
The idea is to have an instance of watcher inside the Cagent struct (`ca.cpuWatcher`). And `GetAllMeasurements()` should be able to retrieve the `cpuWatcher` instance using `ca.CPUWatcher()` in both case of cold run and after

Reply:
I guess this answers my question `Except this makes using it easier in other places`.

What about the errors? Why don't we have to return them?

Reply:
Cpu watcher has different types of measurements inside: different util, loadavg. Some of them could fail. If we return any of errors this will means that we lose the others. More, the error can self-heal after some time(e.g. WMI query on Windows can timeout on the first try)

Instead, we propose to return a watcher in any case. In case of error we just need to log them

Reply:
ok, I see. Makes sense.
I had writing tests in mind... but while thinking about it, testing errors in code that measures CPU stuff might be (too) complicated anyway.
Ok then for the moment